lib/puppet/provider/reboot_notify/notify.rb

```diff
@@ -36,7 +40,11 @@ def update
       :updated => Time.now.tv_sec
     }
 
-    File.open(@target,'w'){|fh| fh.puts(PSON.pretty_generate(@records))}
+    begin
+      File.open(@target,'w'){|fh| fh.puts(JSON.pretty_generate(@records))}
+    rescue
```

Comment:
See comment above about rescuing specific exceptions and including the specifics of that error in the error message.